tools/bridge/bridge/confirmation_sender.py

```diff
@@ -19,6 +20,23 @@
 logger = logging.getLogger(__name__)
 
 
+def make_sanity_check_transfer(foreign_bridge_contract_address):
+    """return a function that checks the final transfer right before it is confirmed
+
+    This is the last safety net. We do check that the transfer has
+    been sent to the foreign_bridge_contract_address here.
+    """
+    assert is_checksum_address(foreign_bridge_contract_address)
+
+    def sanity_check_transfer(transfer_event):
```

Comment:
I like this check. Wondering if we should extend it to also compare the transaction we would send against the event (i.e. if the value and the receiver address matches)?

Reply:
We can certainly add more checks. I wanted to add this after I broke the event filters in the event fetcher.
Other things that also come to mind: protect against Transfers originating from the wrong contract.